src/appleseed/renderer/kernel/lighting/pathtracer.h

```diff
@@ -355,6 +355,14 @@ size_t PathTracer<PathVisitor, VolumeVisitor, Adjoint>::trace(
             continue;
         }
 
+        // Apply OSL holdout.
+        if (material_data.m_shader_group &&
+            material_data.m_shader_group->has_matte() &&
+            vertex.m_path_length == 1)
```

Comment:
Make `vertex.m_path_length == 1` the first condition since it's going to evaluate to false for all bounces but the first, hence short-circuiting the evaluation of the remaining conditions (it's a micro-optimization).